vfs_proxyfs.c

```diff
@@ -1931,9 +1931,35 @@ static int vfs_proxyfs_get_real_filename(struct vfs_handle_struct *handle,
                                          TALLOC_CTX *mem_ctx, char **found_name)
 {
 	DEBUG(10, ("vfs_proxyfs_get_real_filename: %s\n", path));
-	errno = ENOTSUP;
-	return -1;
 
+	if (strlen(name) >= NAME_MAX) {
+		errno = ENAMETOOLONG;
+		return -1;
+	}
+
+	if ((strlen(path) + strlen(name) + 1) > PATH_MAX) {
```

Comment:
I think this should be +2 since snprintf() adds null byte:
     "The functions snprintf() and vsnprintf() write at most size bytes (including the terminating null byte ('\0')) to str."

According to https://linux.die.net/man/3/snprintf

Reply:
Thanks for catching it. Yes, I added it.